### PR TITLE
fix: support x-kubernetes-int-or-string type flag

### LINKED ISSUE #20: the intOrString types in k8s crd is treated as the first type in generation
## Bug Report

the intOrString types in k8s crd is treated as the first type in generation

### 1. Minimal reproduce step (Required)

for ```intOrString``` type declaration, see: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#intorstring

### 2. What did you expect to see? (Required)

generated code: ```attribute: int | str```

### 3. What did you see instead (Required)
generated code: ```attribute: int```

log: ```warning: JSON-Schema type definition as array with several types is not supported in spec.StringOrArray{"integer", "string"}. Taking the first type: integer```

### PR DESCRIPTION
**Pull request by amyXia1994**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):
fix #20 

- [ ] N
- [X] Y 

#### 2. What is the scope of this PR (e.g. component or file name):
pkg/swagger/generator/types.go

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [X] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [X] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [X] Unit test
- [X] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

The test case in pkg/kube_resource/generator/testdata/union_type is used to test the k8s int or str type flag

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
support resolving x-kubernetes-int-or-string extension to KCL int | str type
```

### CODE REVIEW

File: pkg/swagger/generator/types.go [hunk 1]
```diff
@@ -215,6 +216,24 @@ func (t *typeResolver) resolveFormat(schema *spec.Schema, isAnonymous bool, isRe
 	return
 }
 
+func (t *typeResolver) resolveExtensions(schema *spec.Schema, isAnonymous bool, isRequired bool) (returns bool, result resolvedType, err error) {
```
Comment (chai2010) [inline pkg/swagger/generator/types.go:219]: - `resolveExtensions(schema *spec.Schema, isAnonymous, isRequired bool)` second and third args has the same `bool` type, the first type is optional
- return values are confused, what type of return value？`return value, ok` ?

**Review by Peefy — APPROVED** [review]
LGTM

File: pkg/swagger/generator/types.go (hunk 1 above)
Comment (amyXia1994) [reply pkg/swagger/generator/types.go:219]: - fixed, ptal
- there are 3 return values: 
   - ```returns bool```: if the caller function should return. As the OpenAPI spec states, the type resolver will detect the result KCL type by multiple factors, and the type resolver will resolve specific keys step by step. For example if ```$ref``` appears as a key under a property, all the other keys will be ignored and the property's type will be resolved to the referenced model that the "$ref" defined. And in this PR's case, if ```{"x-kubernetes-int-or-string": "true"}``` appears, the type of the property will be ```int | str```, regardless the ```type``` and  ```format``` keys. So in cases above, the main type resolving process has no need to resolve other keys under the property spec and can directly return with the resolved type.
   - ```result resolvedType```: the result of the resolved type
   - ```err error```: if any error occurred in the sub process of resolving extensions, the error should be announced in the main type resolving process

**Review by chai2010 — DISMISSED** [review]
LGTM

**Review by chai2010 — APPROVED** [review]

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**